fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/TempTableInsertPlan.java

```diff
@@ -0,0 +1,227 @@
+        final var tempTable = (TempTable)getTempTableReferenceValue().eval(store, context);
+        tempTable.add(queryResult);
+        return CompletableFuture.completedFuture(queryResult);
+    }
+
+    @Nonnull
+    @Override
+    public TempTableInsertPlan translateCorrelations(@Nonnull final TranslationMap translationMap,
+                                                     @Nonnull final List<? extends Quantifier> translatedQuantifiers) {
+        return new TempTableInsertPlan(
+                Iterables.getOnlyElement(translatedQuantifiers).narrow(Quantifier.Physical.class),
+                getCoercionTrie(),
+                getComputationValue(),
+                getTempTableReferenceValue().translateCorrelations(translationMap));
+    }
+
+    @Nonnull
+    @Override
+    public TempTableInsertPlan withChild(@Nonnull final Reference childRef) {
+        return new TempTableInsertPlan(Quantifier.physical(childRef),
+                getCoercionTrie(),
+                getComputationValue(),
+                getTempTableReferenceValue());
+    }
+
+    @Override
+    public int hashCodeWithoutChildren() {
+        return Objects.hash(BASE_HASH.planHash(PlanHashable.CURRENT_FOR_CONTINUATION), super.hashCodeWithoutChildren());
+    }
+
+    @Override
+    public int planHash(@Nonnull final PlanHashMode mode) {
+        return PlanHashable.objectsPlanHash(mode, BASE_HASH, super.planHash(mode));
+    }
+
+    @Nonnull
+    @Override
+    public String toString() {
+        return PlanStringRepresentation.toString(this);
+    }
+
+    /**
+     * Rewrite the planner graph for better visualization.
+     *
+     * @param childGraphs planner graphs of children expression that already have been computed
+     *
+     * @return the rewritten planner graph that models the filter as a node that uses the expression attribute
+     * to depict the record types this operator filters.
+     */
+    @Nonnull
+    @Override
+    public PlannerGraph rewritePlannerGraph(@Nonnull List<? extends PlannerGraph> childGraphs) {
+
+        final var graphForTarget =
+                PlannerGraph.fromNodeAndChildGraphs(
+                        new PlannerGraph.TemporaryDataNodeWithInfo(getTargetType(), ImmutableList.of(getTempTableReferenceValue().toString())),
+                        ImmutableList.of());
+
+        return PlannerGraph.fromNodeInnerAndTargetForModifications(
+                new PlannerGraph.ModificationOperatorNodeWithInfo(this,
+                        NodeInfo.MODIFICATION_OPERATOR,
+                        ImmutableList.of("TempTableInsert"),
+                        ImmutableMap.of()),
+                Iterables.getOnlyElement(childGraphs), graphForTarget);
+    }
+
+    @Nonnull
+    @Override
+    public PTempTableInsertPlan toProto(@Nonnull final PlanSerializationContext serializationContext) {
+        return PTempTableInsertPlan.newBuilder().setSuper(toRecordQueryAbstractModificationPlanProto(serializationContext)).build();
```

Comment:
You forgot the `tempTableReferenceValue`

Reply:
done.